src/Microsoft.ComponentDetection.Detectors/linux/LinuxScanner.cs

```diff
@@ -135,5 +140,33 @@ public async Task<IEnumerable<LayerMappedLinuxComponents>> ScanLinuxAsync(string
                 return null;
             }
         }
+
+        /// <summary>
+        /// Extracts a package's upstream source name.
+        /// For example some distributions package openssl as openssl-dev, libssl, openssl1.0, etc.
+        /// </summary>
+        private string ExtractPackageSourceName(Package package)
+        {
+            var name = package.Name;
+            switch (package.MetadataType)
+            {
+                case "ApkMetadata":
+                    name = package.Metadata.OriginPackage ?? package.Metadata.Package;
+                    break;
+                case "DpkgMetadata":
+                    name = string.IsNullOrWhiteSpace(package.Metadata.Source)
+                        ? package.Metadata.Package 
+                        : package.Metadata.Source;
+                    break;
+                case "RpmdbMetadata":
+                    name = package.Metadata.Name;
```

Comment:
I don't think we can consistently choose either the binary package name or the source package name for all RPM based distros. Mariner publishes vulnerabilities under source package names, as does SUSE. But RHEL (also used for centos vulns) publish vulnerabilities under the binary package names.

Reply:
Okay, but we just need a more granular decision tree here?

Reply:
Yes. (Though per my email, I'm not sure if component governance might have a need for both binary and source package name.)

Reply:
Given that component-detection doesn't do vulnerability scanning itself, and supports usages other than vulnerability scanning (e.g SBOMs, ad-hoc inventory analysis) I don't think component-detection
-  should discard the binary package name
-  have the API for the Name field be dependent on the current CVE security behaviour of various distros

I've proposed an alternative in https://github.com/microsoft/component-detection/pull/126 which is to keep both binary and source package name.